LoRaEngine/LoraKeysManagerFacade/CreateEdgeDevice.cs

```diff
@@ -82,22 +85,46 @@ public CreateEdgeDevice(RegistryManager registryManager)
             try
             {
                 _ = await this.registryManager.AddDeviceAsync(edgeGatewayDevice);
+                _ = await this.registryManager.AddModuleAsync(new Module(deviceName, "LoRaWanNetworkSrvModule"));
 
-                var deviceConfigurationUrl = Environment.GetEnvironmentVariable("DEVICE_CONFIG_LOCATION");
-                string json = null;
-
-                // todo correct
-                using (var wc = new WebClient())
+                static ConfigurationContent GetConfigurationContent(string configLocation, IDictionary<string, string> tokenReplacements)
                 {
-                    json = wc.DownloadString(deviceConfigurationUrl);
+                    // todo correct
+                    using var wc = new WebClient();
+                    var json = wc.DownloadString(configLocation);
+                    foreach (var r in tokenReplacements)
+                        json = json.Replace(r.Key, r.Value, StringComparison.Ordinal);
+                    return JsonConvert.DeserializeObject<ConfigurationContent>(json);
                 }
 
-                json = ReplaceJsonWithCorrectValues(region, resetPin, json, spiSpeed, spiDev);
+                var deviceConfigurationContent = GetConfigurationContent(Environment.GetEnvironmentVariable("DEVICE_CONFIG_LOCATION"), new Dictionary<string, string>
+                {
+                    ["[$region]"] = region,
+                    ["[$reset_pin]"] = resetPin,
+                    ["[$spi_speed]"] = string.IsNullOrEmpty(spiSpeed) || string.Equals(spiSpeed, "8", StringComparison.OrdinalIgnoreCase) ? string.Empty : ",'SPI_SPEED':{'value':'2'}",
+                    ["[$spi_dev]"] = string.IsNullOrEmpty(spiDev) || string.Equals(spiDev, "2", StringComparison.OrdinalIgnoreCase) ? string.Empty : ",'SPI_DEV':{'value':'1'}"
+                });
+
+                await this.registryManager.ApplyConfigurationContentOnDeviceAsync(deviceName, deviceConfigurationContent);
 
-                var spec = JsonConvert.DeserializeObject<ConfigurationContent>(json);
-                _ = await this.registryManager.AddModuleAsync(new Module(deviceName, "LoRaWanNetworkSrvModule"));
+                if (!string.IsNullOrEmpty(Environment.GetEnvironmentVariable("APPINSIGHTS_INSTRUMENTATIONKEY")))
```

Comment:
I just deployed the template from scratch and we always set APPINSIGHTS_INSTRUMENTATIONKEY. This will always be true. I think you will have to check LOG_ANALYTICS_WORKSPACE_ID instead
![image](https://user-images.githubusercontent.com/8555833/141425807-550abb39-f1c6-4123-bcb3-8c73509b172b.png)

Reply:
Ok I understood from later. let's discuss this

Reply:
As per other thread - will apply this only to monitoring on the edge.

Reply:
4a7615f240aa978ff70b375465e021678bfdde6f